### ISSUE TITLE
Amplify Gen 2: Self-referencing relationship gives error

### ISSUE THREAD

**Issue by mehdi-blanchard** [aws-amplify/amplify-category-api#2716]
### Environment information

```plain text
System:
  OS: Windows 11 10.0.22621
  CPU: (24) x64 12th Gen Intel(R) Core(TM) i7-12850HX
  Memory: 37.68 GB / 63.61 GB
Binaries:
  Node: 22.3.0 - C:\Program Files\nodejs\node.EXE
  Yarn: undefined - undefined
  npm: 10.8.1 - C:\Program Files\nodejs\npm.CMD
  pnpm: undefined - undefined
NPM Packages:
  @aws-amplify/backend: 1.0.0
  @aws-amplify/backend-cli: 1.0.1
  aws-amplify: 6.4.0
  aws-cdk: 2.140.0
  aws-cdk-lib: 2.140.0
  typescript: 5.4.5
AWS environment variables:
  AWS_NODEJS_CONNECTION_REUSE_ENABLED = 1
  AWS_SDK_LOAD_CONFIG = 1
  AWS_STS_REGIONAL_ENDPOINTS = regional
No CDK environment variables
```

### Data packages

```plain text
+-- @aws-amplify/backend-cli@1.0.1
| `-- @aws-amplify/schema-generator@1.0.0
|   `-- @aws-amplify/graphql-schema-generator@0.8.4
`-- @aws-amplify/backend@1.0.0
  `-- @aws-amplify/backend-data@1.0.0
    `-- @aws-amplify/data-construct@1.8.1+-- @aws-amplify/backend-cli@1.0.1
| `-- @aws-amplify/schema-generator@1.0.0
|   `-- @aws-amplify/graphql-schema-generator@0.8.4
`-- @aws-amplify/backend@1.0.0
  `-- @aws-amplify/backend-data@1.0.0
    `-- @aws-amplify/data-construct@1.8.1
```

### Description

When I create a self-referencing relationship, the sandbox gives an error:

```
Account: a
    .model({
      name: a.string().required(),
      parentId: a.id(),
      account: a.belongsTo('Account', 'parentId'),
      children: a.hasMany('Account', 'parentId')
    })
    .authorization((allow) => [allow.authenticated()])
```

**>npx ampx sandbox**

Failed to instantiate data construct
Caused By: Uni-directional relationships are not supported. Add a @belongsTo field in Account to match the @hasMany field Account.children

Resolution: See the underlying error message for more details.

This is the same issue as [https://github.com/aws-amplify/amplify-category-api/issues/2534](url)
It is marked as closed and resolved but the issue still occurs.

Should the self-refencing be done differently?

**Comment by palpatim** [comment]
This was fixed in [@aws-amplify/data-construct@1.8.2](https://github.com/aws-amplify/amplify-category-api/releases/tag/%40aws-amplify%2Famplify-category-api%405.11.4), but your project is still using `@aws-amplify/data-construct@1.8.1` per the report above. Can you do an `npm update` to pull updates to all dependent packages and try again?

**Comment by palpatim** [comment]
Verified this works locally using the latest version. Auth mode is different from the provided example, but that shouldn't impact the schema validation.

```typescript
const schema = a.schema({
  Account: a
    .model({
      name: a.string().required(),
      parentId: a.id(),
      account: a.belongsTo("Account", "parentId"),
      children: a.hasMany("Account", "parentId"),
    })
    .authorization((allow) => [allow.publicApiKey()]),
});

export type Schema = ClientSchema<typeof schema>;

export const data = defineData({
  schema,
  authorizationModes: {
    defaultAuthorizationMode: "apiKey",
    apiKeyAuthorizationMode: { 
      expiresInDays: 30
    }
  },
});
```
